File: sentimental-credit/credit.py
import cs50

number = cs50.get_int("Number: ")
strnum = str(number)


def sumdigit(num):
    sum = 0
    for i in str(num):
        sum += int(i)
    return sum


def check(number):
    sum = 0
    strnumber = str(number)
    n = int(len(strnumber))
    for i in range(n - 2, -1, -2):
        intsnum = int(strnumber[i])
        if len(str(intsnum * 2)) != 1:
            sum += sumdigit(intsnum * 2)
        else:
            sum += intsnum * 2
    for j in range(n - 1, -1, -2):
        sum += int(strnumber[j])
    return sum


strsum = str(check(number))
if strsum[-1] == "0":
    if int(strnum[0]) == 3 and (
        int(strnum[1]) == 4 or int(strnum[1]) == 7 and len(strnum) == 15
    ):
        print("AMEX")
    elif (
        int(strnum[0]) == 5
        and int(strnum[1]) > 0
        and int(strnum[1]) < 6
        and len(strnum) == 16
    ):
        print("MASTERCARD")
    elif int(strnum[0]) == 4 and (len(strnum) == 13 or len(strnum) == 16):
        print("VISA")
    else:
        print("INVALID")
else:
    print("INVALID")

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long n = get_long("Number: ");
    int i = 0;
    long c = n;
    long a = n;
    long b = n;
    int f;
    int sum1 = 0;
    int sum2 = 0;
    int num1;
    int num2;
    int c1;
    int c2;
    while (c > 0)
    {
        c = c / 10;
        i++;
    }
    do
    {
        num1 = a % 10;
        a /= 10;
        sum1 += num1;

        num2 = a % 10;
        a /= 10;
        num2 *= 2;
        if (num2 >= 10)
        {
            c1 = num2 % 10;
            c2 = num2 / 10;
            sum2 += c1 + c2;
        }
        else
        {
            sum2 += num2;
        }
    }
    while (a > 0);
    sum2 = sum1 + sum2;
    printf("sum2 = %i",sum2);
    sum1 = sum2 % 10;
    while (b > 100)
    {
        b /= 10;
    }
    f = b / 10;
    if (sum1 != 0)
    {
        printf("INVALID\n");
    }
    else if (i == 15 && (b == 34 || b == 37))
    {
        printf("AMEX\n");
    }
    else if (i == 16 && (51 <= b && b <= 55))
    {
        printf("MASTERCARD\n");
    }
    else if ((i == 13 || i == 16) && f == 4)
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}
